### ISSUE TITLE
Redline and records Package Notification and Comments - Release

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#3434]
* As an IAO user
* I want to be notified when a redline or records release package is ready to retrieve
* so that I do not have to wait on a loading screen for the job to complete.

**Assumptions & Scope**	
This is related to user stories #3421 and #3231
Both of those user stories should be considered as 'background jobs' - and so we need a notification for the user to know when the job is complete.

What is IN scope?
Notifications for release package export

What is NOT in scope?
Notifications for harms package (see #3433  ) sign-off (needs completion of sign-off user story)

**Acceptance Criteria**	

**Scenario 1: IAO User Exporting Records for Release -   Notify Success**
* GIVEN a IAO user has saved the final records package for download in the Records Review app
* WHEN the job completes, and the PDF is ready
* THEN the IAO assignee and watcher (not ministry) will receive a notification indicating that "Release Package ready for download"

**Scenario 2: IAO User Exporting Records for Release - Notify Fail**
* GIVEN a IAO user has saved the final records package for download in the Records Review app
* WHEN the job has failed
* THEN the IAO assignee and watcher (not ministry) will receive a notification indicating that "Creating the Release Package failed. Please try again"

**Scenario 3: Click 'Release Package notifications'**
* GIVEN a user has received a release package notification
* AND the user has the notification open in the Notification Log
* WHEN the user clicks on the notification
* THEN the user will be directed to the Records Log

**Scenario 4: IAO User Exporting Records for Release - Comment Success**
* GIVEN a IAO user has saved the final records package for download in the Records Review app
* WHEN the job completes, and the PDF is ready
* THEN a system generated comment will be posted indicating that "Release Package ready for download"

**Scenario 5: IAO User Exporting Records for Release - Comment Fail**
* GIVEN a IAO user has saved the final records package for download in the Records Review app
* WHEN the job has failed
* THEN a system generated comment will be posted indicating "Creating the Release Package failed. Please try again"

**Scenario 6: IAO User Creating Redlines -   Notify Success**
* GIVEN a IAO user has saved the redline for download in the Records Review app
* WHEN the job completes, and the PDF is ready
* THEN the assigned and watching ministry an IAO user will receive a notification indicating that "Redlines for sign off ready for download"

**Scenario 7: IAO User Creating Redlines - Notify Fail**
* GIVEN a IAO user has saved the redline for download in the Records Review app
* WHEN the job has failed
* THEN the assigned and watching ministry and IAO user will receive a notification indicating that "Creating the redlines for sign off failed. Please try again"

**Scenario 8: Click redlines notifications**
* GIVEN a user has received redlines notification
* AND the user has the notification open in the Notification Log
* WHEN the user clicks on the notification
* THEN the user will be directed to the Records Log

**Scenario 9: IAO User Creating Redline - Comment Success**
*  GIVEN a IAO user has saved the redline for download in the Records Review app
* WHEN the job completes, and the PDF is ready
* THEN a system generated comment will be posted indicating that "Redlines for sign off ready for download"

**Scenario 10: IAO User Creating Redline - Comment Fail**
*  GIVEN a IAO user has saved the redline for download in the Records Review app
* WHEN the job has failed
* THEN a system generated comment will be posted indicating "Creating the redlines for sign off failed. Please try again"

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
Corresponding Notification Story for release package - of note for both, I created a scenario for failure, as I think that's important to capture if anything goes wrong.

cc: @lmullane @KyEggleston @abin-aot @liseandtea @JHarrietha-AOT

**Comment by lmullane** [comment]
Thanks, @m-prodan. For scenario 2 suggest we use different language than the job failed.

What about something like this:

 "Creating the Release Package failed. Please try again."

**Comment by divyav-aot** [comment]
extra deployment needed: add below mappers to keycload client foi-document-redaction.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBb0JoIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--ace49ba275d731f2d348fc440aa522e4e55171ff/image.png)

**Comment by nimya-aot** [comment]
Will Test this ticket after @divyav-aot fixes the below issue
  https://app.zenhub.com/workspaces/foi-flow-607f0ab6ae54b700111dc733/issues/gh/bcgov/foi-flow/4420

**Comment by nimya-aot** [comment]
It is working as expected , so closing
